### PR TITLE
Implement permission for using SignPortals

### LINKED ISSUE #80: MV-Signportals Permission use of teleportsigns
Hi, I have a suggestion for MV-Signportals.

Would it be possible to implement a permission to allow/deny using of teleportsigns for special groups/users?

Only members of the group/users with this permission can use teleportsigns.

F.e.: multiverse.signportal.use -> allow/deny             (like already existing: multiverse.signportal.create)

This would be very useful. Thx for answering. 🙂 

br slyer9

Quick remark: Denying the use of signs in general would also affect other plugins like "Chestshop", therefore a special permission for MV-Signportals would be fundamental.

### PR DESCRIPTION
**Pull request by zax71**
Fixes #80

### CODE REVIEW

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java [hunk 1]
```diff
@@ -33,14 +33,16 @@
...
+    private static final String USE_PERMISSION = "multiverse.signportal.use";
+    private final MultiverseSignPortals plugin;
+    private final MVPermissions permissions;
+    private final PortalDetector pd;
 
     public MVSPPlayerListener(MultiverseSignPortals plugin) {
         this.plugin = plugin;
         this.permissions = this.plugin.getCore().getMVPerms();
         this.permissions.addPermission("multiverse.signportal.validate", PermissionDefault.OP);
+        this.permissions.addPermission(USE_PERMISSION, PermissionDefault.TRUE);
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:45]: A bit inconsistent to use the constant for one of these and not the other.

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java [hunk 2]
```diff
@@ -83,28 +85,50 @@ public void playerPortal(PlayerPortalEvent event) {
      */
     @EventHandler
     public void playerInteract(PlayerInteractEvent event) {
+        Player player = event.getPlayer();
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:88]: This should be closer to where the actual usage is.

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java [hunk 3]
```diff
@@ -83,28 +85,50 @@ public void playerPortal(PlayerPortalEvent event) {
...
                 }
-            }
+                event.setCancelled(true);
+                break;
+            case Legacy:
+                this.pd.activateSignPortal(event.getPlayer(), ChatColor.AQUA + "Legacy", s);
+                event.setCancelled(true);
+                break;
+            case Disabled:
+                this.pd.activateSignPortal(event.getPlayer(), ChatColor.RED + "Disabled", s);
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:124]: Use the player variable here?

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java [hunk 4]
```diff
@@ -83,28 +86,51 @@ public void playerPortal(PlayerPortalEvent event) {
      */
     @EventHandler
     public void playerInteract(PlayerInteractEvent event) {
+
+
+        // The event must not be canceled...
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:91]: Left some unnecessary whitespace here.

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java [hunk 5]
```diff
@@ -83,28 +86,51 @@ public void playerPortal(PlayerPortalEvent event) {
...
+                break;
+            case Disabled:
+                this.pd.activateSignPortal(player, ChatColor.RED + "Disabled", s);
+                event.setCancelled(true);
+                break;
+            case NetherPortalSign:
+                event.setCancelled(true);
+                break;
+            case NotASignPortal: // We shouldn't bother with other people's stuff
+                event.setCancelled(false);
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:133]: This added case really doesn't need to exist. The event should already not be cancelled so there's really no reason to set it to not cancelled again.

**Review by dumptruckman — CHANGES_REQUESTED** [review]
Getting closer. Just a couple other tidbits.

File: src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java (hunk 5 above)
Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java:133]: Ah, I can remove it, added it there to fill out the Enum mainly. I guess what little Rust I’ve done has affected me 😆

**Review by dumptruckman — APPROVED** [review]
